docs/user_guide.md

```diff
@@ -528,14 +527,38 @@ This section is only applicable to advanced users who wish to create custom
 constraint templates. If the existing templates are sufficient, you can skip
 this section.
 
-#### Template Naming Convention
+#### Template Authoring Convention
+
+##### Naming
+
+The template name appears in three places in a template YAML file:
+
+*   metadata name: All lower case with "-" as the separator. It has the format
+    of "gcp-{resource}-{feature}-{version}" (example: "gcp-storage-logging-v1").
```

Comment:
We don't have the version number as part of the metadata name. I think we need to make this change.